### ISSUE TITLE
Benchmark exception message not reported (edited)

### ISSUE THREAD

**Issue by spawnia** [phpbench/phpbench#916]
Hi, thanks for this library 👋 I am happily using PHPBench to measure the performance of my projects.

Recently, the benchmarks started failing on me: https://github.com/nuwave/lighthouse/pull/1926/checks?check_run_id=3422593650

```
3 subjects encountered errors:

\Benchmarks\HugeResponseBench::benchmark1

    RuntimeException Script "/workdir/vendor/phpbench/phpbench/lib/Executor/Benchmark/template/remote.template" did not return an array, got:

    #0 /workdir/vendor/phpbench/phpbench/lib/Remote/Payload.php(150): PhpBench\Remote\Payload->decodeResults(Object(Symfony\Component\Process\Process))
    #1 /workdir/vendor/phpbench/phpbench/lib/Executor/Benchmark/TemplateExecutor.php(62): PhpBench\Remote\Payload->launch()
```

I tried to remedy the issue by updating to the latest stable version of PHPBench, but found no success: https://github.com/nuwave/lighthouse/pull/1927/checks?check_run_id=3422595605

The offending file seems to be this one: https://github.com/nuwave/lighthouse/blob/master/benchmarks/HugeResponseBench.php. Removing it makes the other benchmark work.

The reason for the failure might be related to an update of underyling dependencies. As Lighthouse is a library,
I don't pin its dependencies.

**Comment by dantleech** [comment]
I've had a quick look and updated with `^1.0.0` using the following config for debugging:

```
{
    "$schema": "./vendor/phpbench/phpbench/phpbench.schema.json",
    "runner.bootstrap": "vendor/autoload.php",
    "runner.path": "benchmarks",
    "runner.remote_script_path":".phpbench/script",
    "runner.remote_script_remove": false,
    "runner.php_config": {
        "memory_limit": -1
    },
    "report.generators": {
        "ast": {
            "extends": "aggregate",
            "title": "AST unserialization",
            "description": "Efficiently unserialize the schema AST",
            "cols": ["subject", "mean", "diff"]
        }
    }
}
```

After running with `--filter=10k` you can then run the script in isolation:

```
php .phpbench/script/remote.template && echo $?
```

It exits without an error in the code which is being benchmarked, I don't think it's due to PHPBench so closing. If you find this is not the case or discover the reason I would be happy to see a resolution :)

**Comment by spawnia** [comment]
Thank you for the feedback. After a bit of trial and error, I was able to pinpoint an error in our test setup and corrected it in https://github.com/nuwave/lighthouse/pull/1927.

However I found that throwing any error within `HugeResponseBench` somehow leads to the aforementioned error message. It seems to be handled properly in another test. Perhaps something can be done about the error reporting? Reproduced the issue here: https://github.com/nuwave/lighthouse/pull/1928

**Comment by dantleech** [comment]
Indeed, the exception message `Auth guard [api] is not defined.` when the script does a try / catch. I'm not sure why there is no error output, but the fix is clear enough.

**Comment by spawnia** [comment]
Figured out why there is no error.

For convenience, I let the `HugeResponseBench` extend the `TestCase`, which contains the necessary `setUp()` to bootstrap the required Laravel environment. As a part of bootstrapping Laravel, I register a custom exception handler that is supposed to rethrow exceptions - see https://github.com/nuwave/lighthouse/blob/05f0eaf06c58291d6207af894599ae1ea5066e0d/tests/Laravel7ExceptionHandler.php.

This worked out fine when running tests, as the bootstrapping also simulates that Laravel is running HTTP requests - thus sending any exceptions to the `render()` method. During benchmarks, it seems that the framework is somehow configured to run as a console application. Thus, exceptions were sent to the `renderForConsole()` method - which was not implemented, as it was not needed before. That caused the exception to be silently discarded and execution to abort without any output, leading to the error I reported in this issue.

The culprit for this was definitely our unusual reuse of the test setup for a benchmark, registering an exception handler that gobbled up all output. I think there is really nothing that PHPBench could have done differently to protect against such misconfiguration and thus consider this issue closed. No bug on your part.

I thank you for taking the time to look into this issue and am happy to continue using PHPBench.
